File: Leetcode/Java/739. Daily Temperatures/solution.java
// https://leetcode.com/problems/daily-temperatures/description/

class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] answer = new int[temperatures.length];
        List<Integer> stack = new ArrayList<>();
        // Update the result if the current temperature warmer
        for (int idx = 0; idx < temperatures.length; idx++){
            while (!stack.isEmpty() && temperatures[idx] > temperatures[stack.get(stack.size() - 1)]){
                int i = stack.remove(stack.size() - 1);
                answer[i] = idx - i;
            }

            stack.add(idx);
        }

        return answer;
    }
}

File: Leetcode/Typescript/739. Daily Temperatures/solution.ts
// https://leetcode.com/problems/daily-temperatures/description/

function dailyTemperatures(temperatures: number[]): number[] {
  const weatherStack: number[] = [],
    result: number[] = Array(temperatures.length).fill(0);

  for (let idx = 0; idx < temperatures.length; idx++) {
    while (
      weatherStack.length > 0 &&
      temperatures[weatherStack[weatherStack.length - 1]] < temperatures[idx]
    ) {
      const recordIndex = weatherStack.pop();
      result[recordIndex!] = idx - recordIndex!;
    }
    weatherStack.push(idx);
  }

  return result;
}
